inventory_management_system_api/repositories/system.py

```diff
@@ -177,20 +177,24 @@ def delete(self, system_id: str, session: ClientSession = None) -> None:
         if result.deleted_count == 0:
             raise MissingRecordError(f"No System found with ID: {str(system_id)}")
 
-    def _is_duplicate_system(self, parent_id: Optional[str], code: str, session: ClientSession = None) -> bool:
+    def _is_duplicate_system(
+        self, parent_id: Optional[str], code: str, system_id: CustomObjectId = None, session: ClientSession = None
+    ) -> bool:
         """
         Check if a System with the same code already exists within the parent System
 
         :param parent_id: ID of the parent System which can also be `None`
         :param code: Code of the System to check for duplicates
+        :param system_id: The ID of the system to check if the duplicate system found is itself.
         :param session: PyMongo ClientSession to use for database operations
         :return: `True` if a duplicate System code is found under the given parent, `False` otherwise
         """
         logger.info("Checking if System with code '%s' already exists within the parent System", code)
         if parent_id:
             parent_id = CustomObjectId(parent_id)
 
-        return self._systems_collection.find_one({"parent_id": parent_id, "code": code}, session=session) is not None
+        system_with_code = self._systems_collection.find_one({"parent_id": parent_id, "code": code}, session=session)
```

Comment:
Calling this `system` should be just fine I think
```suggestion
        system = self._systems_collection.find_one({"parent_id": parent_id, "code": code}, session=session)
```